### PR TITLE
Added Level Up Rewards, UseItemEggIncubator, GetHatchedEggs, UseItemXpBoost (lucky egg) and fixed double firing callbacks.

### PR DESCRIPTION
**Pull request by cokeeffekt**
Callbacks should only be able to fire once, i've fixed up all the callbacks that fire twice. All of them were in try catch blocks. Fired once for try and a second time for catch. 

This commit will allow the ability to use async libraries with out blowing up.

### CODE REVIEW

File: poke.io.js [hunk 1]
```diff
@@ -123,16 +123,15 @@ function Pokeio() {
     };
 
     self.request.post(options, function (err, response, body) {
-      if (err)
-      {
+      if (err) {
```
Comment (fullstackhacker) [inline poke.io.js:126]: :+1:

**Comment by fullstackhacker** [comment]
looks good

**Comment by antonrez89** [comment]
hey, guys, how can I khow the current level of user? I can use LevelUpRewardsMessage and check if RewardStatus  equals AWARDED_ALREADY, but it looks ugly.

**Comment by cokeeffekt** [comment]
@antoxxxa-2 player stats is on the inventory-items object as a entry that has player-stats set.

**Comment by WesVleuten** [comment]
Merging this would be great!

**Comment by Dwynr** [comment]
Whats the correct use for .UseItemEggIncubator()?

I am getting "Error: Illegal value for [object Object] of type string: number (not a string)" everytime I try it with the Incubator Id (901/902 I believe) and the egg Id formatted as a string.

**Comment by Dwynr** [comment]
{ Error: Missing at least one required field for Message .ResponseEnvelop.UseItemEggIncubatorResponse: egg_incubator

**Comment by Jamby93** [comment]
Could anyone push this commit to npm repository? I would really like to have this try-catch fix but I would like to stick to npm dependency instead of cloning from github.

**Pull request merged**